docs/credhub-security-faq.md

```diff
@@ -51,13 +51,29 @@ TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
 ## How are Keys internally managed?
 
 If CredHub's third-party integration with a Luna SafeNet HSM is
-enabled, then CredHub has no knowledge of key values. If CredHub is
-started in internal encryption mode, CredHub deterministically
-generates a site-specific AES256 key on startup. CredHub concatenates
-a user-defined password from its configuration file with a
-randomly-generated salt stored in its database and hashes the
-resulting string to deterministically generate an AES256 key, which it
-holds in memory for the lifetime of the server.
+enabled, then CredHub has no knowledge of key values.
+
+If CredHub is started in internal encryption mode:
+
+* If a new encryption key is provided, CredHub will generate a new
+  random salt string, and will store it in its database for the
+  future.
+* If the encryption key had already been used before, CredHub will
+  fetch the associated salt from its database to reuse it.
+
+CredHub then calculates the AES256 key by concatenating the
+user-provided encryption password from its configuration file with the
+salt, and hashes the resulting string. This hash is then used as the
+AES key, which CredHub will hold in memory.
+
+This design allows CredHub to have access to the AES key at all times,
+without ever having to store it on disk.
+
+Passwords and salts cannot be rotated individually, and they are never
```

Comment:
This could probably be simplified. What people most likely care about is the the salt that can't be rotated by itself. This already explains that the salt will be changed when the password is changed. Maybe -

```
The password and salt are never automatically rotated. Also, the salt cannot be rotated by itself without changing the password.
```